java/core/src/java/org/apache/orc/impl/ConvertTreeReaderFactory.java

```diff
@@ -1764,6 +1812,9 @@ public void nextVector(ColumnVector previousVector,
                                                    " proleptic Gregorian dates.");
           }
         }
+      } else {
+        bytesColVector.ensureSize(batchSize, false);
+        dateColumnVector.ensureSize(batchSize, false);
```

Comment:
`dateColumnVector.ensureSize(batchSize, false);` may throw a null pointer exception and we may need to check additionally.

Reply:
`dateColumnVector` can be ignored. This is because it is the vector obtained by forcing `previousVector` .

https://github.com/apache/orc/blob/3eac56761f48f3e315eea56dc63de9f2cbadcdbb/java/core/src/java/org/apache/orc/impl/RecordReaderImpl.java#L1423

`computeBatchSize` ensures that batchSize <= the size of the input vector.

Reply:
handled it, thanks!

Reply:
> `dateColumnVector` can be ignored. This is because it is the vector obtained by forcing `previousVector` .
> 
> https://github.com/apache/orc/blob/3eac56761f48f3e315eea56dc63de9f2cbadcdbb/java/core/src/java/org/apache/orc/impl/RecordReaderImpl.java#L1423
> 
> `computeBatchSize` ensures that batchSize <= the size of the input vector.

@abstractdog, in addition to this, LGTM

Reply:
okay, removed
thanks a lot for the quick reviews @guiyanakuang !